### DISCUSSION TITLE
ks0108 with U8G2 and TcMenu 4.1.2

### DISCUSSION THREAD

**Discussion by reza6707-reza6707** [davetcc/tcMenu#435] [General]
hi every one .
how can i generate code for ks0108 in TcMenu 4.1.2 software ???

**Comment by davetcc** [comment]
This should be a discussion, please do not create issues that do not follow the issue template. 

If U8G2 supports it, then you can use it with the custom U8G2 plugin. See the custom section of the U8G2 plugin docs:

https://www.thecoderscorner.com/products/arduino-libraries/tc-menu/tcmenu-plugins/u8g2-renderer-plugin/#manual-configuration

**Reply by reza6707-reza6707** [reply]
tnx for your reply.
but
in TcMenu app U8g2 used for any Display with I2C or SPI .
Ks0108 use parallel wiring for exchnge data between arduino .
How can i generated code from TcMenu to my program .

**Reply by davetcc** [reply]
So, let's break this down. You want to use KS0108 with tcMenu. Firstly, you need to check if your configuration is supported by any of the display libraries that tcMenu supports. It seems U8G2 has support.

If it is supported by one of the libraries, then I would recommend that you first test your display is working with an example sketch. Once you've confirmed this and got it working with a supported library then you could use the "Custom U8G2" plugin.

1. create a menu project with a couple of items in it
2. generate the code and choose the "Custom U8G2" plugin, this allows YOU to define the display variable. In the display parameters set the display variable to the name of the variable YOU will create in the sketch, and also set the display type to the type of that variable. This is so TcMenu can create an `extern` declaration for it.
3. in your sketch, take the working U8G2 variable and define it globally, then add the setup code into your setup method.
